delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -0,0 +1,144 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch
+
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewCommand._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewEvent._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewRejection._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewState.{Current, Initial}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewValue._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.{IOUtils, UUIDF}
+import io.circe.Json
+import monix.bio.{IO, UIO}
+
+class ElasticSearchViews {}
+
+object ElasticSearchViews {
+
+  /**
+    * The elasticsearch module type.
+    */
+  final val moduleType: String = "elasticsearchview"
+
+  private[elasticsearch] def next(
+      state: ElasticSearchViewState,
+      event: ElasticSearchViewEvent
+  ): ElasticSearchViewState = {
+    // format: off
+    def created(e: ElasticSearchViewCreated): ElasticSearchViewState = state match {
+      case Initial     => Current(e.id, e.project, e.uuid, e.value, e.source, Map.empty, e.rev, deprecated = false,  e.instant, e.subject, e.instant, e.subject)
+      case s: Current  => s
+    }
+
+    def updated(e: ElasticSearchViewUpdated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, value = e.value, source = e.source, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    def tagAdded(e: ElasticSearchViewTagAdded): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, tags = s.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+    }
+    // format: on
+
+    def deprecated(e: ElasticSearchViewDeprecated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    event match {
+      case e: ElasticSearchViewCreated    => created(e)
+      case e: ElasticSearchViewUpdated    => updated(e)
+      case e: ElasticSearchViewTagAdded   => tagAdded(e)
+      case e: ElasticSearchViewDeprecated => deprecated(e)
+    }
+  }
+
+  private[elasticsearch] def evaluate(
+      validatePermission: Permission => IO[PermissionIsNotDefined, Unit],
+      validateMapping: Json => IO[InvalidElasticSearchMapping, Unit],
```

Comment:
As explained later, it would be nice to have a ElasticsearchMapping type, that would change this method to: 
```validateMapping: Json => IO[InvalidElasticSearchMapping, ElasticSearchMapping]```

Reply:
I tried this locally, it doesn't work very well with the current code. We just need to make sure the value is validated; after that it doesn't make much use since we're not going to do anything else with the value.